src/supertux/autotile_parser.cpp

```diff
@@ -196,17 +196,27 @@ AutotileParser::parse_autotile(const ReaderMapping& reader, bool corner)
 
       alt_id += m_offset;
 
-      float weight = 0.0f;
-      if (!alt_reader.get("weight", weight))
-      {
-        log_warning << "No weight for alt tile id" << std::endl;
+      if (!alt_id)
         continue;
-      }
 
-      if (alt_id != 0 && weight != 0.0f)
+      std::vector<uint32_t> period_x = { 0, 0 };
+      std::vector<uint32_t> period_y = { 0, 0 };
+      float weight = 0.f;
+      alt_reader.get("period-x", period_x);
+      alt_reader.get("period-y", period_y);
+      alt_reader.get("weight", weight);
+
+      if (period_x.size() != 2 || period_y.size() != 2)
       {
-        alt_ids.push_back(std::pair<uint32_t, float>(alt_id, weight));
+        log_warning << "X/Y period for alt tile must contain exactly 2 values" << std::endl;
+        continue;
       }
+
+      alt_ids.push_back(std::pair<uint32_t, Autotile::AltConditions>(alt_id, {
+          { period_x[0], period_x[1] },
+          { period_y[0], period_y[1] },
+          weight
+        }));
```

Comment:
Does this mean you can put alt-ids with no conditions now?

Reply:
Yes. If you do, they just won't do anything.